### ISSUE TITLE
NuGet Add Packages - FormatException when offline

### ISSUE THREAD

**Issue by mrward** [mono/monodevelop#7498]
This error is logged and no information is reported in the UI.

```
An unhandled exception has occurred. Terminating Visual Studio? False
System.FormatException: Input string was not in a correct format.
  at System.Text.StringBuilder.AppendFormatHelper (System.IFormatProvider provider, System.String format, System.ParamsArray args) [0x000b2] in /Users/builder/jenkins/workspace/build-package-osx-mono/2018-08/external/bockbuild/builds/mono-x64/external/corefx/src/Common/src/CoreLib/System/Text/StringBuilder.cs:1445 
  at System.String.FormatHelper (System.IFormatProvider provider, System.String format, System.ParamsArray args) [0x0000e] in /Users/builder/jenkins/workspace/build-package-osx-mono/2018-08/external/bockbuild/builds/mono-x64/external/corefx/src/Common/src/CoreLib/System/String.Manipulation.cs:471 
  at System.String.Format (System.String format, System.Object[] args) [0x00018] in /Users/builder/jenkins/workspace/build-package-osx-mono/2018-08/external/bockbuild/builds/mono-x64/external/corefx/src/Common/src/CoreLib/System/String.Manipulation.cs:436 
  at MonoDevelop.PackageManagement.AllPackagesViewModel.NuGet.PackageManagement.UI.INuGetUILogger.Log (NuGet.ProjectManagement.MessageLevel level, System.String message, System.Object[] args) [0x00004] in /Users/vsts/agent/2.150.3/work/1/s/monodevelop/main/src/addins/MonoDevelop.PackageManagement/MonoDevelop.PackageManagement/AllPackagesViewModel.cs:520 
  at NuGet.PackageManagement.UI.MultiSourcePackageFeed+<>c__DisplayClass15_0.<LogError>b__0 () [0x00011] in /Users/vsts/agent/2.150.3/work/1/s/monodevelop/main/src/addins/MonoDevelop.PackageManagement/NuGet.PackageManagement.UI/MultiSourcePackageFeed.cs:250 
  at MonoDevelop.Core.Runtime+<>c.<RunInMainThread>b__42_0 (System.Object state) [0x00013] in /Users/vsts/agent/2.150.3/work/1/s/monodevelop/main/src/core/MonoDevelop.Core/MonoDevelop.Core/Runtime.cs:340 
```

The problem is that the NSUrlSessionHandler exception message uses curly braces which the string.format tries to process and fails.

```
[nuget.org] The Internet connection appears to be offline.
  Error Domain=NSURLErrorDomain Code=-1009 "The Internet connection appears to be offline." UserInfo={NSUnderlyingError=0x7fce42892a80 {Error Domain=kCFErrorDomainCFNetwork Code=-1009 "(null)" UserInfo={_kCFStreamErrorCodeKey=50, _kCFStreamErrorDomainKey=1}}, NSErrorFailingURLStringKey=https://api-v2v3search-1.nuget.org/query?q=&skip=0&take=26&prerelease=false&supportedFramework=Xamarin.iOS,Version=v1.0&semVerLevel=2.0.0, NSErrorFailingURLKey=https://api-v2v3search-1.nuget.org/query?q=&skip=0&take=26&prerelease=false&supportedFramework=Xamarin.iOS,Version=v1.0&semVerLevel=2.0.0, _kCFStreamErrorDomainKey=1, _kCFStreamErrorCodeKey=50, NSLocalizedDescription=The Internet connection appears to be offline.}
```

> VS bug [#889359](https://devdiv.visualstudio.com/DevDiv/_workitems/edit/889359)

**Comment by xamarin-release-manager** [comment]
> [sync] [VS-13] Comment by **xamarinc**



Fixed in version 8.3.0.111 (master)



Pull Request #7500 merged by: Lluis Sanchez  
 Author: mono  
 Commit: 1639171ef392a1fc1c1e01637ca3c961592efb83 (mono/monodevelop)

**Comment by xamarin-release-manager** [comment]
> [sync] [VS-14] Comment by **xamarinc**



Fixed in version 8.2.0.542 (release-8.2)



Pull Request #7500 merged by: Lluis Sanchez  
 Author: mono  
 Commit: 1639171ef392a1fc1c1e01637ca3c961592efb83 (mono/monodevelop)

**Comment by xamarin-release-manager** [comment]
> [sync] [VS-19] Comment by **xamarinc**



Fixed in version 8.6.0.802 (master)



Pull Request #7500 merged by: Lluis Sanchez  
 Author: mono  
 Commit: 1639171ef392a1fc1c1e01637ca3c961592efb83 (xamarin/monodevelop)
